### ISSUE TITLE
Issue lab_2

### ISSUE THREAD

**Issue by CappePolito** [GioSilve/CI2024_lab2#3]
The code solves the TSP problem in a really good way, these are the main focus points i will focus on:

Greedy start: The initialization with the greedy start chooses a starting city and successively selecting the nearest unvisited city, until all of them are covered. It's very efficient, but might get stuck in a local optimal. Euristics like nearest neighbor might randomize a bit more the starting popoulation, but greedy is still very good

PMX: Partially mapped crossover is very effective for the tsp problem.

Inversion mutation: It's very effective in tsp problems, good choice

Parent selection: The parent selection uses a tournament selection strategy, selecting the fitter individual from 2 randomly chosen cities. It's a good choice in genetic algorithms problems, and it's easy to implement

Notes:
Usually in genetic algorithm problems keeping the fittest individuals (elites) with the elitist approach usually is common and might help with faster convergence rates.

The number of generation is fixed: it could be set maybe to stop after x generations with no to little improvement to better minimize the cost

In short: good job, the code is clear and well written, it's efficient and well optimized, and i can only suggest little improvements!
The final cost is a bit high, but maybe with more generations it will improve (mine took 3.5 hours to reach 54000, so i understand that you didnt want to wait for so long XD)
